presto-kafka/src/main/java/io/prestosql/plugin/kafka/confluent/AvroConfluentRowDecoderFactory.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kafka.confluent;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.prestosql.decoder.DecoderColumnHandle;
+import io.prestosql.decoder.RowDecoder;
+import io.prestosql.decoder.RowDecoderFactory;
+import org.apache.avro.Schema;
+
+import javax.inject.Inject;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.util.Objects.requireNonNull;
+
+public class AvroConfluentRowDecoderFactory
+        implements RowDecoderFactory
+{
+    private final Optional<SchemaRegistryClient> schemaRegistryClient;
+
+    @Inject
+    public AvroConfluentRowDecoderFactory(Optional<SchemaRegistryClient> schemaRegistryClient)
```

Comment:
If we have a implementation like [this](https://github.com/prestosql/presto/pull/2499/files?file-filters%5B%5D=.java&file-filters%5B%5D=.rst#r522085211). Then we `SchemaRegistryClient` and `AvroConfluentRowDecoderFactory` will be binded on demand. We don't have to inject `SchemaRegistryClient` as an Optional